mlrun/runtimes/pod.py

```diff
@@ -660,6 +661,51 @@ def kube_resource_spec_to_pod_spec(
     )
 
 
+def enrich_resources_with_default_pod_resources(resources: dict = None):
+    resources_types = ["cpu", "memory", "gpu"]
+    resource_requirements = ["requests", "limits"]
+    default_resources = mlconf.default_function_pod_resources.to_dict()
+    if not resources:
+        return default_resources
+
+    _verify_gpu_requests_and_limits(
+        requests_gpu=resources.setdefault("requests", {}).setdefault("gpu", ""),
+        limits_gpu=resources.setdefault("requests", {}).setdefault("gpu", ""),
+    )
+    for resource_requirement in resource_requirements:
+        for resource_type in resources_types:
+            if not resources.setdefault(resource_requirement, {}).setdefault(
+                resource_type, ""
+            ):
+                if resource_type == "gpu" and resource_requirement == "requests":
+                    # set gpu defaults only on limits
+                    continue
+
+                resources[resource_requirement][resource_type] = default_resources[
+                    resource_requirement
+                ][resource_type]
+
+            if resources[resource_requirement][resource_type]:
+                verify_field_regex(
+                    field_name=f"function.spec.resources.{resource_requirement}.{resource_type}",
+                    field_value=resources[resource_requirement][resource_type],
+                    patterns=mlrun.utils.regex.k8s_resource_quantity_regex,
+                )
```

Comment:
I feel like we're unneeded-ly duplicating the verification we already have in `with_limits` and `with_requests` any reason not to use them ?